Week 4 Hangman/main.py

```diff
@@ -129,47 +129,59 @@ def correctGuess(cls,correctGuesses,word) -> list[str]:
         print("")
         return lst
     
-    @classmethod
-    def incorrectGuess(cls) -> None:
+    def incorrectGuess(self) -> None:
         print("Incorrect")
 
 class Game:
-    word = RandomWordGenerator.generate()
-    guessCount = GuessCount.generate(word)
-    wordGuessed = False
-    partialGuessedString = []
-    def __init__(self) -> None:
-        pass
     
-    @classmethod
-    def play(cls) -> None:
+    def __init__(self) -> None:
+        self._printString = PrintString()
+        self._guessCheck = GuessCheck()
+        self._guessCount = GuessCount().generate()
+        self._word = RandomWordGenerator().generate()
+        self._wordGuessed = False
+        self._user = User()
+        self._partialGuessedString = []
```

Comment:
Two things here:
1. The name of the variable (`_partialGuessedString`) does not agree with its type (`list`).  Perhaps `_partially_guessed_word` might be better?
2. What is the purpose of this member, and why is it a list in the first place?

Reply:
Purpose of this member is to check if player has already guessed the word and end the game even if his guesses are not yet used up